src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/Internal/Json/MessageTypeInfoResolver.cs

```diff
@@ -59,18 +59,18 @@ public MessageTypeInfoResolver(JsonContext context)
         return typeInfo;
     }
 
-    private static bool IsStandardMessage(Type type, [NotNullWhen(true)] out MessageDescriptor? messageDescriptor)
+    private bool IsStandardMessage(Type type, [NotNullWhen(true)] out MessageDescriptor? messageDescriptor)
     {
         if (!typeof(IMessage).IsAssignableFrom(type))
         {
             messageDescriptor = null;
             return false;
         }
 
-        messageDescriptor = JsonConverterHelper.GetMessageDescriptor(type);
+        messageDescriptor = (MessageDescriptor?) _context.DescriptorRegistry.FindDescriptorByType(type);
         if (messageDescriptor == null)
         {
-            return false;
+            throw new InvalidOperationException("Couldn't resolve descriptor for message type: " + type);
```

Comment:
Where is this caught?  Does it end up triggering a telemetry event or something?

Reply:
(If I'm reading it correctly, it indicates a programmer error (e.g. rather than malformed input).)

Reply:
It will be caught during the serialization of a request/response.

This error means something has gone unexpectedly wrong in storing or resolving descriptors from the registry.